sdk/core/azure-core-tracing-opentelemetry/src/samples/PublishEventsTracingSample.md

```diff
@@ -80,35 +80,63 @@ public class Sample {
+            // We will publish three events based on simple sentences.
+            Flux<EventData> data = Flux.just(
+                new EventData(body).addContext(PARENT_SPAN_KEY, TRACER.getCurrentSpan()),
+                new EventData(body2).addContext(PARENT_SPAN_KEY, TRACER.getCurrentSpan()));
+
+            // Create a batch to send the events.
             final CreateBatchOptions options = new CreateBatchOptions()
-                .setPartitionKey("telemetry")
+                .setPartitionId(firstPartition)
                 .setMaximumSizeInBytes(256);
-    
-            EventDataBatch currentBatch = producer.createBatch(options);
-    
-            // For each telemetry event, we try to add it to the current batch.
-            for (EventData event : telemetryEvents) {
-                if (!currentBatch.tryAdd(event)) {
-                    producer.send(currentBatch);
-                    currentBatch = producer.createBatch(options);
+
+            final AtomicReference<EventDataBatch> currentBatch = new AtomicReference<>(
+                producer.createBatch(options).block());
+            
+            data.flatMap(event -> {
+                final EventDataBatch batch = currentBatch.get();
+                if (batch.tryAdd(event)) {
+                    return Mono.empty();
                 }
-            }
-        } finally {
+
+                // The batch is full, so we create a new batch and send the batch. Mono.when completes when both operations
+                // have completed.
+                return Mono.when(
+                        producer.send(batch),
+                        producer.createBatch(options).map(newBatch -> {
+                            currentBatch.set(newBatch);
+
+                            // Add that event that we couldn't before.
+                            if (!newBatch.tryAdd(event)) {
+                                throw Exceptions.propagate(new IllegalArgumentException(String.format(
+                                        "Event is too large for an empty batch. Max size: %s. Event: %s",
+                                        newBatch.getMaxSizeInBytes(), event.getBodyAsString())));
+                            }
+
+                            return newBatch;
+                        }));
+            }).then()
+                    .doFinally(signal -> {
+                        final EventDataBatch batch = currentBatch.getAndSet(null);
+                        if (batch != null) {
+                            producer.send(batch).block(OPERATION_TIMEOUT);
+                        }
+                    })
+                    .subscribe(unused -> System.out.println("Complete"),
+                            error -> System.out.println("Error sending events: " + error),
+                            () -> System.out.println("Completed sending events."));
             span.end();
-            producer.close();
```

Comment:
We should close the producer when the send is complete. And same with the span in do finally.